### PR TITLE
feat(serverless-hub): Update to enable generalize bot runner to work in production

### PR DESCRIPTION
**Pull request by chrismaree**
**Motivation**

We have built a set of tools internally that enable us to run a series of bots using a "strategy runner" with minimal config. Using this, a bot runner can define a very succinct config that is expanded to run n number of bots.

Ideally, this setup should also work within our serverless infrastructure to simplify bot running internally.

This PR the strategy runner and some logic in the serverless hub to facilitate this. 

**Summary**

There are three main changes in this PR:
- Modified how the main winston `Transports` module works with google `trace-agent` to first check if the agent has started before initializing the module. If this is _not_ done running the strategy runner in production errors out on child process execution.
- Modified the bot strategy runner to support config objects added within stringified environment variables. This helps with running within production cloud run setups.
- Modified the serverlessHub to include a `commonConfig` param that enables the hub to define a set of common fields that should be injected into all spoke processes. This will enable common fields (such as whitelists for bot strategy runner) to be defined _once_ at the top of the config file and then re-used be each config.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [X]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/bot-strategy-runner/src/index.ts [hunk 1]
```diff
@@ -118,13 +118,18 @@ const processExecutionOptions = async () => {
     .option("-fc, --fileConfig <path>", "input path to JSON config file.")
     .option("-uc, --urlConfig <path>", "url to JSON config hosted online. Private resources use access token")
     .option("-at, --accessToken <string>", "access token to access private configs online. EG private a repo")
+    .option("-env, --envConfig <bool>", "use a stringified config within your env set under `RUNNER_CONFIG")
     .parse(process.argv)
     .opts();
 
-  assert(options.fileConfig != undefined || options.urlConfig != undefined, "provide a file config or a URL config");
+  assert(
+    options.fileConfig != undefined || options.urlConfig != undefined || options.envConfig,
```
Comment (nicholaspai) [inline packages/bot-strategy-runner/src/index.ts:126]: why not write this as `assert(options.fileConfig || options.urlConfig || options.envConfig)`?

File: packages/financial-templates-lib/src/logger/Transports.js [hunk 2]
```diff
@@ -18,15 +18,15 @@ function createTransports(transportsConfig = {}) {
   // If the logger is running in serverless mode then add the GCP winston transport and console transport.
   if ((transportsConfig.environment ?? process.env.ENVIRONMENT) == "serverless") {
     const { LoggingWinston } = require("@google-cloud/logging-winston");
-    require("@google-cloud/trace-agent").start();
+    if (!require("@google-cloud/trace-agent").get().enabled) require("@google-cloud/trace-agent").start();
```
Comment (daywiss) [inline packages/financial-templates-lib/src/logger/Transports.js:21]: this library is imported mulitple times, is there a reason you dont just import once?

File: packages/bot-strategy-runner/src/index.ts [hunk 3]
```diff
@@ -118,13 +118,18 @@ const processExecutionOptions = async () => {
     .option("-fc, --fileConfig <path>", "input path to JSON config file.")
     .option("-uc, --urlConfig <path>", "url to JSON config hosted online. Private resources use access token")
     .option("-at, --accessToken <string>", "access token to access private configs online. EG private a repo")
+    .option("-env, --envConfig <bool>", "use a stringified config within your env set under `RUNNER_CONFIG")
```
Comment (daywiss) [inline packages/bot-strategy-runner/src/index.ts:121]: why do u need to set this option explicltly, why not just merge the env RUNNER_CONFIG if it exists?

**Review by daywiss — APPROVED** [review]
lgtm

File: packages/bot-strategy-runner/src/index.ts (hunk 1 above)
Comment (chrismaree) [reply packages/bot-strategy-runner/src/index.ts:126]: fair! have refactored

File: packages/bot-strategy-runner/src/index.ts (hunk 3 above)
Comment (chrismaree) [reply packages/bot-strategy-runner/src/index.ts:121]: just to be more explicit.

Comment (chrismaree) [reply packages/bot-strategy-runner/src/index.ts:121]: but you are right. I'll modify this sligtly to let it pull this in by default without the user having to say they are using an env config.

File: packages/financial-templates-lib/src/logger/Transports.js (hunk 2 above)
Comment (daywiss) [reply packages/financial-templates-lib/src/logger/Transports.js:21]: i dont see an easy way to do that, unless its instanced at the top level and passed in as a dependency, but its not worth the effort.

File: packages/bot-strategy-runner/src/index.ts (hunk 3 above)
Comment (daywiss) [reply packages/bot-strategy-runner/src/index.ts]: i could imagine you might not want to delete some complex env json, but maybe disable it instead so you can easily enable it later. might be a use case for that, its up to you.

**Pull request merged**